rpc/jsonrpc/server/http_uri_handler.go

```diff
@@ -56,8 +58,17 @@ func makeHTTPHandler(rpcFunc *RPCFunc, logger log.Logger) func(http.ResponseWrit
 		logger.Debug("HTTPRestRPC", "method", r.URL.Path, "args", args, "returns", returns)
 		result, err := unreflectResult(returns)
 		if err != nil {
-			WriteRPCResponseHTTPError(w, http.StatusInternalServerError,
-				types.RPCInternalError(dummyID, err))
+			switch errors.Unwrap(err) {
```

Comment:
This doesn't seem to be working 😞 

I wonder if it has anything to do with the reflect package. It just isn't able to unwrap the errors. I will have to look into this a bit more

Reply:
Okay yeah I believe when we reflect to the `error` type it gets essentially squashed so that there is nothing to unwrap.

My solution around this would be to use `strings.Contains(err.Error(), ctypes.ErrZeroOrNegativeHeight.Error())`
i.e. simple string matching. We just need to make sure that no error is a subset (in terms of strings) of another error which shouldn't be too hard to do.

The other thing I could experiment with is having error structs that implement the error interface and see if we can use those to be able to categorize the type of error

Reply:
I did it!! don't know how but it works
![](https://media.giphy.com/media/doPrWYzSG1Vao/giphy.gif)

Reply:
Perfect! Clean code 👍